ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,305 @@
+
+        Args:
+           name (str): The Underlying Storage name to be created
+           region (str): The region to create the Underlying Storage, if none will create at
+           `us-east-1` IMPORTANT!!! note that `us-east-1` will cause an error if used since it is
+           the default region for aws
+
+        """
+        if region is None:
+            self.client.create_bucket(Bucket=name)
+        else:
+            self.client.create_bucket(
+                Bucket=name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+    def internal_delete_uls(self, name):
+        """
+        Deletes the Underlying Storage using the S3 API
+
+        Args:
+           name (str): The Underlying Storage name to be deleted
+
+        """
+        # Todo: rename client to resource (or find an alternative)
+        self.client.meta.client.delete_bucket_policy(
+            Bucket=name
+        )
+        sample = TimeoutSampler(
+            timeout=30, sleep=3, func=self.check_uls_deletion,
+            name=name
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(
+                f'Deletion of Underlying Storage {name} timed out. Unable to delete {name}'
+            )
+            raise TimeoutExpiredError
+        else:
+            logger.info(f'Underlying Storage {name} deleted successfully')
+
+    def check_uls_deletion(self, name):
+        """
+        Try to delete Underlying Storage by name
+
+        Args:
+            name (str): the Underlying Storage name
+
+        Returns:
+            bool: True if deleted successfully
+
+        """
+        if self.verify_uls_exists(name):
+            try:
+                self.client.Bucket(name).objects.all().delete()
+                self.client.Bucket(name).delete()
+                return True
+            except ClientError:
+                logger.info(f'Deletion of Underlying Storage {name} failed. Retrying...')
+                return False
+        else:
+            return True
+
+    def get_all_uls_names(self):
+        return {bucket.name for bucket in self.client.buckets.all()}
+
+    def verify_uls_exists(self, uls_name):
+        """
+       Verifies whether a Underlying Storage with the given uls_name exists
+
+       Args:
+           uls_name (str): The Underlying Storage name to be verified
+
+       Returns:
+             bool: True if Underlying Storage exists, False otherwise
+
+        """
+        try:
+            # Todo: rename client to resource (or find an alternative)
+            self.client.meta.client.head_bucket(Bucket=uls_name)
+            logger.info(f"{uls_name} exists")
+            return True
+        except ClientError:
+            logger.info(f"{uls_name} does not exist")
+            return False
+
+    def get_oc_secret(self):
+        return self.secret.name
+
+    def get_aws_key(self):
```

Comment:
Any reason we're using those getters instead of directly accessing the members?
[#1]